modules/apps/asset/asset-list-web/src/main/java/com/liferay/asset/list/web/internal/portlet/action/UpdateAssetListEntryDynamicMVCActionCommand.java

```diff
@@ -116,14 +116,20 @@ private AssetQueryRule _getQueryRule(
 				actionRequest, "queryTagNames" + index);
 		}
 		else if (name.equals("keywords")) {
-			StrTokenizer strTokenizer = new StrTokenizer(
-				ParamUtil.getString(actionRequest, "keywords" + index));
-
-			strTokenizer.setQuoteMatcher(StrMatcher.quoteMatcher());
-
-			List<String> valuesList = (List<String>)strTokenizer.getTokenList();
-
-			values = valuesList.toArray(new String[0]);
+			values = TransformUtil.transformToArray(
+				_keywordTokenizer.tokenize(
+					ParamUtil.getString(actionRequest, "keywords" + index)),
+				value -> {
+					if ((value.length() > 2) &&
+						(value.charAt(0) == CharPool.QUOTE) &&
+						(value.charAt(value.length() - 1) == CharPool.QUOTE)) {
+
+						return value.substring(1, value.length() - 1);
+					}
+
+					return value;
+				},
+				String.class);
```

Comment:
@tinatian I tried this on a test program to see the difference between common-langs StrTokenizer and search team's SimpleKeywordTokenizer. The result is this:
```
Test String: "Hello this \"is an example\""
common-langs tokenize with quote matcher: [Hello, this, is an example]
search keyword tokenizer: [Hello, this, "is an example"]
```